junit-platform-testkit/src/main/java/org/junit/platform/testkit/engine/Events.java

```diff
@@ -336,4 +398,60 @@ private static void assertEventsMatchExactly(List<Event> events, Condition<? sup
 		softly.assertAll();
 	}
 
+	@SafeVarargs
+	private static void assertEventsMatchLoosely(List<Event> events, Condition<? super Event>... conditions) {
+		SoftAssertions softly = new SoftAssertions();
+		for (Condition<? super Event> condition : conditions)
+			checkCondition(events, softly, condition);
+		softly.assertAll();
+	}
+
+	@SafeVarargs
+	@SuppressWarnings("varargs")
+	private static void assertEventsMatchIncompleteButOrdered(List<Event> events,
+			Condition<? super Event>... conditions) {
+		Assertions.assertThat(conditions).hasSizeLessThanOrEqualTo(events.size());
+		SoftAssertions softly = new SoftAssertions();
+
+		// @formatter:off
+		List<Integer> indices = Arrays.stream(conditions)
+				.map(condition -> findEvent(events, softly, condition))
+				.filter(Objects::nonNull)
+				.map(events::indexOf)
+				.collect(toList());
+		// @formatter:on
+
+		if (isNotInIncreasingOrder(indices))
+			softly.fail("Conditions are not in the correct order.");
+
+		softly.assertAll();
+	}
+
+	private static boolean isNotInIncreasingOrder(List<Integer> indices) {
+		List<Integer> copy = new ArrayList<>(indices);
+		sort(copy);
+
+		return !indices.equals(copy);
+	}
+
+	private static void checkCondition(List<Event> events, SoftAssertions softly, Condition<? super Event> condition) {
+		boolean matches = events.stream().anyMatch(condition::matches);
+
+		if (!matches)
+			softly.fail("Condition did not match any event: " + condition);
```

Comment:
Same as above regarding `softly.fail(..)`.

Reply:
ditto

Reply:
This one seems to be different though: The soft assertion can indeed transport the information that more than one condition did not match any event.
So I am leaving this case as is.